File: COData-WEB-Desarrollo-isai/FrontEnd-VS-Code/COData-Angular/src/app/models/reportes.ts
import { Usuario } from './usuario';
import { Categorias } from './categorias';
import { Ubicaciones } from './ubicaciones';

export interface Reportes {
  reporteId: number;
  usuarioId: number;
  categoriaId: number;
  ubicacionId: number;
  titulo: string;
  descripcion: string;
  estado: string;
  prioridad: string;
  fecha_Creacion: Date;
  fecha_Cierre?: Date | null;
  usuario?: Usuario;
  categoria?: Categorias;
  ubicacion?: Ubicaciones;
}

File: COData-WEB-Desarrollo-isai/BackEnd-VS-C#/COData-Web_BackEnd/COData-Web_BackEnd/Models/Reportes.cs
namespace COData_Web_BackEnd.Models
{
    public class Reportes
    {
        public int ReporteId { get; set; }
        public int UsuarioId { get; set; }
        public int CategoriaId { get; set; }
        public int UbicacionId { get; set; }
        public string Titulo { get; set; } = string.Empty;
        public string Descripcion { get; set; } = string.Empty;
        public string Estado { get; set; } = string.Empty;
        public string Prioridad { get; set; } = string.Empty;
        public DateTime Fecha_Creacion { get; set; }
        public DateTime? Fecha_Cierre { get; set; }
        // Relaciones
        public Usuario Usuario { get; set; } = new();
        public Categorias Categoria { get; set; } = new();
        public Ubicaciones Ubicacion { get; set; } = new();
    }
}
